src/cli.ts

```diff
@@ -112,33 +104,15 @@ function debugCliInfo(version: string, channel: string, env: Env, config: Interf
   );
 }
 
-type CreateOptions = {
-  version: string;
-  bin: string | undefined;
-  channel: string;
-  run?: typeof oclifRun;
-  env?: typeof nodeEnv;
-};
-
-export function create({ version, bin, channel, run, env }: CreateOptions): { run: () => Promise<unknown> } {
-  settings.performanceEnabled = true;
-  const root = resolve(fileURLToPath(import.meta.url), '..');
-  const args = process.argv.slice(2);
-  const environment = env ?? nodeEnv;
-  return {
-    async run(): Promise<unknown> {
-      const config = new Config({
-        name: bin,
-        root,
-        version,
-        channel,
-      });
-      await config.load();
-      configureUpdateSites(config, environment);
-      configureAutoUpdate(environment);
-      debugCliInfo(version, channel, environment, config);
-      // Example of how run is used in a test https://github.com/salesforcecli/cli/pull/171/files#diff-1deee0a575599b2df117c280da319f7938aaf6fdb0c04bcdbde769dbf464be69R46
-      return run ? run(args, config) : oclifRun(args, config);
-    },
-  };
+export async function run(): Promise<unknown> {
```

Comment:
this is the part I don't understand (why we were doing it the other way before, what changed to make this the better option).  It looks much cleaner, though!